Lib/functools.py

```diff
@@ -889,13 +889,19 @@ def register(cls, func=None):
 
             # only import typing if annotation parsing is necessary
             from typing import get_type_hints
-            argname, cls = next(iter(get_type_hints(func).items()))
+            from annotationlib import Format, ForwardRef
+            argname, cls = next(iter(get_type_hints(func, format=Format.FORWARDREF).items()))
```

Comment:
We could possibly fix #110373 at the same time here by using `typing.evaluate_forward_ref()` instead of `get_type_hints`. `singledispatch` only needs to evaluate the first type hint on a function; anything else is doing extra work and leads to bugs

Reply:
I think I'll leave that for another PR, added it to #119180.